app/services/foreman_salt/fact_parser.rb

```diff
@@ -88,17 +88,12 @@ def support_interfaces_parsing?
 
     def os_hash
       name = facts[:os]
-      (_, major, minor, sub) = /(\d+)\.?(\d+)?\.?(\d+)?/.match(facts[:osrelease]).to_a
-      minor = "" if minor.nil?
-      if name == 'CentOS Stream'
-        return { :name => name.tr(' ', '_'), :major => major, :minor => minor }
-      end
-      if name == 'CentOS' || name == 'CentOS Linux'
-        if sub
-          minor += '.' + sub
-        end
-        return { :name => 'CentOS', :major => major, :minor => minor }
-      end
+      name = 'CentOS' if ['CentOS', 'CentOS Stream', 'CentOS Linux'].include?(name)
+
+      match = /(?<major>\d+)(\.?(?<minor>\d+(\.\d+)?))?/.match(facts[:osrelease])
+      major = match[:major]
+      minor = match[:minor] || ""
```

Comment:
I think this is a good improvement regardless of anything else and I now realize technically this refactor belongs in its own commit.